tools/dynamic_framework_slicer/dynamic_framework_slicer.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash 
+set -eu
+
+BINARIES+=()
+while [[ $# -gt 0 ]]; do
+  arg="$1"
+  case $arg in
+      --in)
+      readonly IN="$2"
+      shift
+      ;;
+      --out)
+      readonly OUT="$2"
+      shift
+      ;;
+      --realpath)
+      readonly REALPATH="$2"
+      shift
+      ;;
+      *)
+      BINARIES+=("$1")
+      ;;
+  esac
+  shift
+done
+
+# Strip out any unnecessary slices from embedded dynamic frameworks to save space
+
+for bin in $BINARIES; do
+    # Lets figure out what slices the app has
+    bin_slices=$(xcrun lipo -info "$bin" | cut -d: -f3 | sort)
+
+    if [ -f "$IN" ]; then
+        # We need to check we have overlapping architectures
+        framework_slices=$(xcrun lipo -info "$IN" | cut -d: -f3 |sort)
+        
+        # The normal case should be that the framework and app have exactly the same slices.
+        # Lets check that and skip processing if that is the case
+        if [ "$bin_slices" == "$framework_slices" ]; then
+            continue;
```

Comment:
Does the Starlark expect to find the framework in the output dir at the end, that doesn't happen in this case, no?  i.e. - if there is nothing to strip, support break?

Reply:
Good call.